### PR TITLE
Add JSON encoders, string and JSON representation for various classes

### PR DESCRIPTION
**Pull request by bhufmann**
- Add JSON encoders, string and JSON representation for 
  - Identifier
  - Configuration related classes
  - Experiment and traces
  Use this to print identifier as JSON string. **This will increase the readability of the command-line results.**
- Replace to_string with string representation method for Health class
- Remove unused constant and comment in tsp_cli_client
- Add list-output example to README.md

To test the print-outs invoke the corresponding CLI commands.

Signed-off-by: Bernd Hufmann <bernd.hufmann@ericsson.com>

### CODE REVIEW

File: tsp/configuration.py [hunk 1]
```diff
@@ -71,11 +73,23 @@ def __init__(self, params):
...
+        return json.dumps(self, cls=ConfigurationEncoder, indent=4)
+
+
+class ConfigurationEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, Configuration):
+            return {
+                ID_KEY: obj.id,
+                NAME_KEY: obj.name,
+                DESCTIPION_KEY: obj.description,
```
Comment (PatrickTasse) [inline tsp/configuration.py:90]: Constant should be renamed to DESCRIPTION_KEY

File: tsp/configuration_source.py [hunk 2]
```diff
@@ -71,19 +72,27 @@ def __init__(self, params):
...
-    # pylint: disable=consider-using-f-string
-    def to_string(self):
-        '''
-        to_string 
-        '''
-        my_str = "no parameter descriptor"
-        if self.parameter_descriptors is not None:
-            my_str = self.parameter_descriptors.to_string()
+    def __repr__(self):
+        return 'ConfigurationSource(id={}, name={}, description={}, parameter_descriptor={}, schema={})'.format(
```
Comment (PatrickTasse) [inline tsp/configuration_source.py:76]: parameter_descriptors

File: tsp_cli_client [hunk 3]
```diff
@@ -483,17 +483,22 @@ if __name__ == "__main__":
                 if not configuration_source_set or len(configuration_source_set.configuration_source_set) == 0:
                     print('No configuration sources available')
                 else:
-                    print('  {0}'.format(configuration_source_set.to_string()))
+                    print('Successfully listed configuration sources')
+                    print('-----------------------------------------')
+                    print(response.model.to_json())
                 sys.exit(0)
             else:
                 sys.exit(1)
```
Comment (PatrickTasse) [inline tsp_cli_client:467]: Just wondering why we only print message on error in certain cases?

**Review by PatrickTasse — CHANGES_REQUESTED** [review]

File: tsp/configuration.py (hunk 1 above)
Comment (bhufmann) [reply tsp/configuration.py:90]: Done

File: tsp/configuration_source.py (hunk 2 above)
Comment (bhufmann) [reply tsp/configuration_source.py:76]: Done

File: tsp_cli_client (hunk 3 above)
Comment (bhufmann) [reply tsp_cli_client:467]: The error is printed in the library. For this case see `tsp_client.fetch_configuration_sources()`

**Review by PatrickTasse — APPROVED** [review]

**Pull request merged**